hs_app_timeseries/page_processors.py

```diff
@@ -1,106 +1,221 @@
 
+    extended_metadata_exists = False
+    if content_model.metadata.sites or \
+            content_model.metadata.variables or \
+            content_model.metadata.methods or \
+            content_model.metadata.processing_levels or \
+            content_model.metadata.time_series_results:
+        extended_metadata_exists = True
+
+    series_ids = {}
+    for result in content_model.metadata.time_series_results:
+        series_id = result.series_ids[0]
+        series_ids[series_id] = _get_series_label(series_id, content_model)
+
+    if 'series_id' in request.GET:
+        selected_series_id = request.GET['series_id']
+        if 'series_id' in request.session:
+            if selected_series_id != request.session['series_id']:
+                is_resource_specific_tab_active = True
+                request.session['series_id'] = selected_series_id
+            else:
+                is_resource_specific_tab_active = False
+        else:
+            request.session['series_id'] = selected_series_id
+            is_resource_specific_tab_active = False
+    else:
+        selected_series_id = series_ids.keys()[0] if series_ids.keys() else None
+        request.session['series_id'] = selected_series_id
+        is_resource_specific_tab_active = False
+
     # view depends on whether the resource is being edited
     if not edit_resource:
-        # READONLY
-        # get the context from hs_core
-        context = page_processors.get_page_context(page, request.user, resource_edit=edit_resource, extended_metadata_layout=None, request=request)
-        extended_metadata_exists = False
-        if content_model.metadata.site or \
-                content_model.metadata.variable or \
-                content_model.metadata.method or \
-                content_model.metadata.processing_level or \
-                content_model.metadata.time_series_result:
-            extended_metadata_exists = True
-
-        context['extended_metadata_exists'] = extended_metadata_exists
-        context['site'] = content_model.metadata.site
-        context['variable'] = content_model.metadata.variable
-        context['method'] = content_model.metadata.method
-        context['processing_level'] = content_model.metadata.processing_level
-        context['timeseries_result'] = content_model.metadata.time_series_result
+        # resource in VIEW Mode
+        context = _get_resource_view_context(page, request, content_model, selected_series_id,
+                                             series_ids, extended_metadata_exists)
     else:
-        # EDIT MODE
+        # resource in EDIT Mode
+        context = _get_resource_edit_context(page, request, content_model, selected_series_id,
+                                             series_ids, extended_metadata_exists)
+
+    context['is_resource_specific_tab_active'] = is_resource_specific_tab_active
+
+    # TODO: can we refactor to make it impossible to skip adding the generic context
```

Comment:
The comment makes a good point but have not figured out yet how we can achieve that.